b9a879034e introduces new build-time warning in utf8.c

When building perl today on FreeBSD-14, where `clang18` is the default compiler, encountered this build-time warning:
```
utf8.c:2322:47: warning: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
 2322 |                                               MIN(avail_len,
      |                                               ^   ~~~~~~~~~
 2323 |                                                   overlong_detect_length),
      |                                                   ~~~~~~~~~~~~~~~~~~~~~~
/usr/include/sys/param.h:331:23: note: expanded from macro 'MIN'
  331 | #define MIN(a,b) (((a)<(b))?(a):(b))
      |                     ~ ^ ~
./embed.h:1802:80: note: expanded from macro 'byte_dump_string_'
 1802 | #   define byte_dump_string_(a,b,c)             Perl_byte_dump_string_(aTHX_ a,b,c)
      |                                                                                ^
1 warning generated.
```
Running `git blame utf8.c` suggests that this warning began to be emitted with [this Sept 19 commit](b9a879034eb):
```
commit b9a879034ebd7cae703557bf58e9c42e3712686c
Author:     Karl Williamson <khw@cpan.org>
AuthorDate: Sat Jan 18 10:27:39 2025 -0700
Commit:     Karl Williamson <khw@cpan.org>
CommitDate: Fri Sep 19 07:58:43 2025 -0600

    utf8_to_uv_msgs: Clarify warning message
    
    Previously, this output an error message consisting of two characters
    while calling it a single malformed character.  And croak.t tested for
    that.  But that message is misleading.

diff --git a/t/lib/croak/toke_l1 b/t/lib/croak/toke_l1
...
diff --git a/utf8.c b/utf8.c
index 90ec770264..722195b4e7 100644
--- a/utf8.c
+++ b/utf8.c
@@ -1978,6 +1978,11 @@ Perl_utf8_to_uv_msgs_helper_(const U8 * const s0,
      *                      end, based on how many bytes the start byte tells
      *                      us should be in it, but no further than s0 +
      *                      avail_len
+     * overlong_detect_length  if no overlong malformation is present, this is
+     *                      0; otherwise it is the number of bytes required to
+     *                      make that determination.  It is used below to limit
+     *                      the number of bytes displayed in a warning so as to
+     *                      make the warning accurate and not misleading.
      */
     bool success = true;
 
@@ -2296,8 +2301,11 @@ Perl_utf8_to_uv_msgs_helper_(const U8 * const s0,
                             " \"%s\" is overlong which can and should be"
```
@khwilliamson, can you take a look?  Thanks.

--- a/embed.fnc
+++ b/embed.fnc
@@ -6179,7 +6179,7 @@ RTi	|int	|does_utf8_overflow					\
 				|NN const U8 *e
 RTi	|int	|isFF_overlong	|NN const U8 * const s			\
 				|const STRLEN len
-RTi	|int	|is_utf8_overlong					\
+RTi	|SSize_t|is_utf8_overlong					\
 				|NN const U8 * const s			\
 				|const STRLEN len
 RS	|HV *	|new_msg_hv	|NN const char * const message		\

--- a/proto.h
+++ b/proto.h
@@ -9661,7 +9661,7 @@ S_isFF_overlong(const U8 * const s, const STRLEN len)
 #   define PERL_ARGS_ASSERT_ISFF_OVERLONG       \
         assert(s)
 
-PERL_STATIC_INLINE int
+PERL_STATIC_INLINE SSize_t
 S_is_utf8_overlong(const U8 * const s, const STRLEN len)
         __attribute__warn_unused_result__;
 #   define PERL_ARGS_ASSERT_IS_UTF8_OVERLONG    \

--- a/utf8.c
+++ b/utf8.c
@@ -461,7 +461,7 @@ The new names accurately describe the situation in all cases.
 =cut
 */
 
-PERL_STATIC_INLINE int
+PERL_STATIC_INLINE SSize_t
 S_is_utf8_overlong(const U8 * const s, const STRLEN len)
 {
     /* Returns an int indicating whether or not the UTF-8 sequence from 's' to
@@ -649,7 +649,7 @@ S_does_utf8_overflow(const U8 * const s, const U8 * e)
   overflows_if_not_overlong: ;
 
     /* Here, the sequence overflows if not overlong.  Check for that */
-    int is_overlong = is_utf8_overlong(s, len);
+    SSize_t is_overlong = is_utf8_overlong(s, len);
     if (LIKELY(is_overlong == 0)) {
         return OVERFLOWS;
     }
@@ -2319,8 +2319,8 @@ Perl_utf8_to_uv_msgs_helper_(const U8 * const s0,
                             malformed_text,
                             byte_dump_string_(s0, curlen, 0),
                             byte_dump_string_(s0,
-                                              MIN(avail_len,
-                                                  overlong_detect_length),
+                                              MIN((SSize_t) avail_len,
+                                                   overlong_detect_length),
                                               0));
                 }
                 else {

